### ISSUE TITLE
Deserialization fails if scans is empty

### ISSUE THREAD

**Issue by mganss** [Genbox/VirusTotal.NET#17]
This code

```C#
  //Warning: Huge hack... sorry :(
response.Content = Regex.Replace(response.Content, "\"([\\w\\d -\\._]+)\": \\{\"detected\":", "{\"name\": \"$1\", \"detected\":", RegexOptions.Compiled | RegexOptions.CultureInvariant);
response.Content = response.Content.Replace("scans\": {", "scans\": [");
response.Content = response.Content.Replace("}}", "}]");
```

results in invalid JSON if scans is {}. The result is `"scans": [}]` which is caused by the last line that assumes the scans object to always contain a member.

To fix this you could add this as the first line:

```
response.Content = response.Content.Replace("scans\": {}", "scans\": []");
```

Another option would be to change the type of the Scans property from `List<ScanEngine>` to `Dictionary<string, ScanEngine>`. This would allow you to remove the hack.

On a side note, I'm not sure if it makes sense to retry the request if the deserialízation fails. In most cases, like this one, it will only hide bugs on the client side.

**Comment by Genbox** [comment]
Thanks for reporting the issue.

Regarding the hack, In order for Dictonary<string, ScanEngine> to work, I would still need to do the hack to convert scans to a set, and do some regex magic to get RestSharp to recognize the name of the engines as strings instead of classes. Do you have another solution in mind?

As for the retries, I found that VT sometimes return invalid JSON (empty scans is just one of them). It seems that retrying the query against VT often fixes the issue (I suspect it might be a concurrency bug), which is why I retry on a SerializationException.

The most pressing issue I see with the current solution, is that a retry takes up a HTTP request. Alternatively, I would let the SerializationException fall though and let the user handle it. Any ideas on this?

**Comment by mganss** [comment]
Deserializing from a JSON object to a `Dictionary<string,XYZ>` "just works". The only problem (besides it being a breaking change) is that then the `Name` properties of the `ScanEngine` objects are not set (they are the deserialized to the keys of the dictionary). You could either remove the Name property or set it after deserialization:

```C#
foreach (var kvp in ...) kvp.Value.Name = kvp.Key;
```

What other kinds of invalid JSON have you seen? The empty scans is not technically invalid JSON, so if this bug is fixed, it should no longer trigger an exception.

I'd let the exception fall through because, as you said, the retries take up HTTP requests. If, on the other hand, rate limiting support would be built in to the library, ... :wink:

**Comment by Genbox** [comment]
Could you show me the code that deserializes into a Dictonary<string, ScanEngine>?

You are right about empty scans not being invalid JSON. It is an invalid API response as they guarantee to always return a response code. I will handle the empty response better and remove the retry logic.

I've seen results with unescaped " and others in the data, resulting in exceptions. Most of them are not fixed by retrying, which is why handling the empty scans case is better and then let the exception fall through.

**Comment by mganss** [comment]
```C#
public class MyUrlReport
{
    /// <summary>
    /// Filescan Id of the resource.
    /// </summary>
    public string FilescanId { get; set; }

    /// <summary>
    /// A permanent link that points to this specific scan.
    /// </summary>
    public string Permalink { get; set; }

    /// <summary>
    /// How many engines flagged this resource.
    /// </summary>
    public int Positives { get; set; }

    /// <summary>
    /// Contains the id of the resource. Can be a SHA256, MD5 or other hash type.
    /// </summary>
    public string Resource { get; set; }

    /// <summary>
    /// The response code. Use this to determine the status of the report.
    /// </summary>
    public ReportResponseCode ResponseCode { get; set; }

    /// <summary>
    /// The date the resource was last scanned.
    /// </summary>
    public DateTime ScanDate { get; set; }

    /// <summary>
    /// Contains the scan id for this result.
    /// </summary>
    public string ScanId { get; set; }

    /// <summary>
    /// The scan results from each engine.
    /// </summary>
    public Dictionary<string, ScanEngine> Scans { get; set; }

    /// <summary>
    /// How many engines scanned this resource.
    /// </summary>
    public int Total { get; set; }

    /// <summary>
    /// Contains the message that corrosponds to the reponse code.
    /// </summary>
    public string URL { get; set; }

    /// <summary>
    /// Contains the message that corrosponds to the reponse code.
    /// </summary>
    public string VerboseMsg { get; set; }
}
```

and then

```C#
var json = @"[{""permalink"": ""https://www.virustotal.com/url/bb0413b30da127632d2fc03ddbb676655737d462f6ef952426e057f83164ca80/analysis/1445560681/"", ""resource"": ""http://static.updatestar.net/dl/realtek/usbcardreader.exe"", ""url"": ""http://static.updatestar.net/dl/realtek/usbcardreader.exe"", ""response_code"": 1, ""scan_date"": ""2015-10-23 00:38:01"", ""scan_id"": ""bb0413b30da127632d2fc03ddbb676655737d462f6ef952426e057f83164ca80-1445560681"", ""verbose_msg"": ""Scan finished, scan information embedded in this object"", ""filescan_id"": ""2e98cab6901eef871649baa202fd6dd7a0783bcc5d74bd2ff709189046ac3d9e-1445560685"", ""positives"": 1, ""total"": 65, ""scans"": {""CLEAN MX"": {""detected"": false, ""result"": ""clean site""}, ""Rising"": {""detected"": false, ""result"": ""clean site""}, ""OpenPhish"": {""detected"": false, ""result"": ""clean site""}, ""VX Vault"": {""detected"": false, ""result"": ""clean site""}, ""ZDB Zeus"": {""detected"": false, ""result"": ""clean site""}, ""AutoShun"": {""detected"": false, ""result"": ""unrated site""}, ""ZCloudsec"": {""detected"": false, ""result"": ""clean site""}, ""PhishLabs"": {""detected"": false, ""result"": ""unrated site""}, ""Zerofox"": {""detected"": false, ""result"": ""clean site""}, ""K7AntiVirus"": {""detected"": false, ""result"": ""clean site""}, "" CloudStat"": {""detected"": false, ""result"": ""clean site""}, ""SecureBrain"": {""detected"": false, ""result"": ""clean site""}, ""Quttera"": {""detected"": false, ""result"": ""clean site""}, ""AegisLab WebGuard"": {""detected"": false, ""result"": ""clean site""}, ""MalwareDomainList"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://www.malwaredomainlist.com/mdl.php?search=static.updatestar.net""}, ""ZeusTracker"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""https://zeustracker.abuse.ch/monitor.php?host=static.updatestar.net""}, ""zvelo"": {""detected"": false, ""result"": ""clean site""}, ""Google Safebrowsing"": {""detected"": false, ""result"": ""clean site""}, ""FraudScore"": {""detected"": false, ""result"": ""clean site""}, ""Kaspersky"": {""detected"": false, ""result"": ""clean site""}, ""BitDefender"": {""detected"": false, ""result"": ""clean site""}, ""Wepawet"": {""detected"": false, ""result"": ""clean site""}, ""G-Data"": {""detected"": false, ""result"": ""clean site""}, ""C-SIRT"": {""detected"": false, ""result"": ""clean site""}, ""CyberCrime"": {""detected"": false, ""result"": ""clean site""}, ""Websense ThreatSeeker"": {""detected"": false, ""result"": ""clean site""}, ""MalwarePatrol"": {""detected"": false, ""result"": ""clean site""}, ""Webutation"": {""detected"": false, ""result"": ""clean site""}, ""Trustwave"": {""detected"": false, ""result"": ""clean site""}, ""Web Security Guard"": {""detected"": false, ""result"": ""clean site""}, ""ADMINUSLabs"": {""detected"": false, ""result"": ""clean site""}, ""Malwarebytes hpHosts"": {""detected"": false, ""result"": ""clean site""}, ""Dr.Web"": {""detected"": true, ""result"": ""malicious site""}, ""AlienVault"": {""detected"": false, ""result"": ""clean site""}, ""Emsisoft"": {""detected"": false, ""result"": ""clean site""}, ""Malc0de Database"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://malc0de.com/database/index.php?search=static.updatestar.net""}, ""SpyEyeTracker"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""https://spyeyetracker.abuse.ch/monitor.php?host=static.updatestar.net""}, ""malwares.com URL checker"": {""detected"": false, ""result"": ""clean site""}, ""Phishtank"": {""detected"": false, ""result"": ""clean site""}, ""Malwared"": {""detected"": false, ""result"": ""clean site""}, ""Avira"": {""detected"": false, ""result"": ""clean site""}, ""StopBadware"": {""detected"": false, ""result"": ""unrated site""}, ""Antiy-AVL"": {""detected"": false, ""result"": ""clean site""}, ""SCUMWARE.org"": {""detected"": false, ""result"": ""clean site""}, ""FraudSense"": {""detected"": false, ""result"": ""clean site""}, ""Opera"": {""detected"": false, ""result"": ""clean site""}, ""Comodo Site Inspector"": {""detected"": false, ""result"": ""clean site""}, ""Malekal"": {""detected"": false, ""result"": ""clean site""}, ""ESET"": {""detected"": false, ""result"": ""clean site""}, ""Sophos"": {""detected"": false, ""result"": ""unrated site""}, ""Yandex Safebrowsing"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://yandex.com/infected?l10n=en&url=http://static.updatestar.net/dl/realtek/usbcardreader.exe""}, ""Spam404"": {""detected"": false, ""result"": ""clean site""}, ""Malware Domain Blocklist"": {""detected"": false, ""result"": ""clean site""}, ""Blueliv"": {""detected"": false, ""result"": ""clean site""}, ""Netcraft"": {""detected"": false, ""result"": ""unrated site""}, ""PalevoTracker"": {""detected"": false, ""result"": ""clean site""}, ""CRDF"": {""detected"": false, ""result"": ""clean site""}, ""ThreatHive"": {""detected"": false, ""result"": ""clean site""}, ""ParetoLogic"": {""detected"": false, ""result"": ""clean site""}, ""Tencent"": {""detected"": false, ""result"": ""clean site""}, ""URLQuery"": {""detected"": false, ""result"": ""unrated site""}, ""Sucuri SiteCheck"": {""detected"": false, ""result"": ""clean site""}, ""Fortinet"": {""detected"": false, ""result"": ""clean site""}, ""ZeroCERT"": {""detected"": false, ""result"": ""clean site""}, ""Baidu-International"": {""detected"": false, ""result"": ""clean site""}}}, {""permalink"": ""https://www.virustotal.com/url/91cab30accfdbfc2f8210dd80339049ea2e70e27499c743b0f5ad7ae701a91c2/analysis/1371373847/"", ""resource"": ""http://software-files-a.cnet.com/s/software/12/28/73/50/cbsidlm-tr1_13-CyberLink_Power2Go_Essential-BP-84589.exe?lop=link&ptype=3001&ontid=2646&siteId=4&edId=3&spi=9e82fb74a571fa1f7c82153b3c662f7f&pid=12287350&psid=84589&isDlm=1&token=1369854425_306a0e73e9047ad1efced658feb027d7&fileName=cbsidlm-tr1_13-CyberLink_Power2Go_Essential-BP-84589.exe"", ""url"": ""http://software-files-a.cnet.com/s/software/12/28/73/50/cbsidlm-tr1_13-CyberLink_Power2Go_Essential-BP-84589.exe?lop=link&ptype=3001&ontid=2646&siteId=4&edId=3&spi=9e82fb74a571fa1f7c82153b3c662f7f&pid=12287350&psid=84589&isDlm=1&token=1369854425_306a0e73e9047ad1efced658feb027d7&fileName=cbsidlm-tr1_13-CyberLink_Power2Go_Essential-BP-84589.exe"", ""response_code"": 1, ""scan_date"": ""2013-06-16 09:10:47"", ""scan_id"": ""91cab30accfdbfc2f8210dd80339049ea2e70e27499c743b0f5ad7ae701a91c2-1371373847"", ""verbose_msg"": ""Scan finished, scan information embedded in this object"", ""filescan_id"": null, ""positives"": 0, ""total"": 0, ""scans"": {}}, {""permalink"": ""https://www.virustotal.com/url/843233860e8a7508da6c00d0e16bb7fedf7ef1e74071fedced1a7357263d7965/analysis/1447411301/"", ""resource"": ""http://www.teamviewer.com/download/TeamViewer_Setup.exe"", ""url"": ""http://www.teamviewer.com/download/TeamViewer_Setup.exe"", ""response_code"": 1, ""scan_date"": ""2015-11-13 10:41:41"", ""scan_id"": ""843233860e8a7508da6c00d0e16bb7fedf7ef1e74071fedced1a7357263d7965-1447411301"", ""verbose_msg"": ""Scan finished, scan information embedded in this object"", ""filescan_id"": ""668ba2180a8c010320a275cb09191cf477993993022bd73387c01d0f66784664-1447364680"", ""positives"": 0, ""total"": 66, ""scans"": {""CLEAN MX"": {""detected"": false, ""result"": ""clean site""}, ""Rising"": {""detected"": false, ""result"": ""clean site""}, ""OpenPhish"": {""detected"": false, ""result"": ""clean site""}, ""VX Vault"": {""detected"": false, ""result"": ""clean site""}, ""ZDB Zeus"": {""detected"": false, ""result"": ""clean site""}, ""AutoShun"": {""detected"": false, ""result"": ""unrated site""}, ""ZCloudsec"": {""detected"": false, ""result"": ""clean site""}, ""PhishLabs"": {""detected"": false, ""result"": ""unrated site""}, ""Zerofox"": {""detected"": false, ""result"": ""clean site""}, ""K7AntiVirus"": {""detected"": false, ""result"": ""clean site""}, ""SecureBrain"": {""detected"": false, ""result"": ""clean site""}, ""Quttera"": {""detected"": false, ""result"": ""clean site""}, ""AegisLab WebGuard"": {""detected"": false, ""result"": ""clean site""}, ""MalwareDomainList"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://www.malwaredomainlist.com/mdl.php?search=www.teamviewer.com""}, ""ZeusTracker"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""https://zeustracker.abuse.ch/monitor.php?host=www.teamviewer.com""}, ""zvelo"": {""detected"": false, ""result"": ""clean site""}, ""Google Safebrowsing"": {""detected"": false, ""result"": ""clean site""}, ""FraudScore"": {""detected"": false, ""result"": ""clean site""}, ""Kaspersky"": {""detected"": false, ""result"": ""clean site""}, ""BitDefender"": {""detected"": false, ""result"": ""clean site""}, ""Opera"": {""detected"": false, ""result"": ""clean site""}, ""G-Data"": {""detected"": false, ""result"": ""clean site""}, ""C-SIRT"": {""detected"": false, ""result"": ""clean site""}, ""CyberCrime"": {""detected"": false, ""result"": ""clean site""}, ""Websense ThreatSeeker"": {""detected"": false, ""result"": ""clean site""}, ""MalwarePatrol"": {""detected"": false, ""result"": ""clean site""}, ""Webutation"": {""detected"": false, ""result"": ""clean site""}, ""Trustwave"": {""detected"": false, ""result"": ""clean site""}, ""Web Security Guard"": {""detected"": false, ""result"": ""clean site""}, ""ADMINUSLabs"": {""detected"": false, ""result"": ""clean site""}, ""Malwarebytes hpHosts"": {""detected"": false, ""result"": ""clean site""}, ""Dr.Web"": {""detected"": false, ""result"": ""clean site""}, ""AlienVault"": {""detected"": false, ""result"": ""clean site""}, ""Emsisoft"": {""detected"": false, ""result"": ""clean site""}, ""Malc0de Database"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://malc0de.com/database/index.php?search=www.teamviewer.com""}, ""SpyEyeTracker"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""https://spyeyetracker.abuse.ch/monitor.php?host=www.teamviewer.com""}, ""malwares.com URL checker"": {""detected"": false, ""result"": ""clean site""}, ""Phishtank"": {""detected"": false, ""result"": ""clean site""}, ""CloudStat"": {""detected"": false, ""result"": ""clean site""}, ""Malwared"": {""detected"": false, ""result"": ""clean site""}, ""Avira"": {""detected"": false, ""result"": ""clean site""}, ""StopBadware"": {""detected"": false, ""result"": ""unrated site""}, ""Antiy-AVL"": {""detected"": false, ""result"": ""clean site""}, ""SCUMWARE.org"": {""detected"": false, ""result"": ""clean site""}, ""FraudSense"": {""detected"": false, ""result"": ""clean site""}, ""Comodo Site Inspector"": {""detected"": false, ""result"": ""clean site""}, ""Malekal"": {""detected"": false, ""result"": ""clean site""}, ""ESET"": {""detected"": false, ""result"": ""clean site""}, ""Sophos"": {""detected"": false, ""result"": ""unrated site""}, ""Yandex Safebrowsing"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://yandex.com/infected?l10n=en&url=http://www.teamviewer.com/download/TeamViewer_Setup.exe""}, ""Spam404"": {""detected"": false, ""result"": ""clean site""}, ""Nucleon"": {""detected"": false, ""result"": ""clean site""}, ""Malware Domain Blocklist"": {""detected"": false, ""result"": ""clean site""}, ""Blueliv"": {""detected"": false, ""result"": ""clean site""}, ""Netcraft"": {""detected"": false, ""result"": ""unrated site""}, ""PalevoTracker"": {""detected"": false, ""result"": ""clean site""}, ""CRDF"": {""detected"": false, ""result"": ""clean site""}, ""ThreatHive"": {""detected"": false, ""result"": ""clean site""}, ""ParetoLogic"": {""detected"": false, ""result"": ""clean site""}, ""Tencent"": {""detected"": false, ""result"": ""clean site""}, ""URLQuery"": {""detected"": false, ""result"": ""unrated site""}, ""Sucuri SiteCheck"": {""detected"": false, ""result"": ""clean site""}, ""Wepawet"": {""detected"": false, ""result"": ""clean site""}, ""Fortinet"": {""detected"": false, ""result"": ""clean site""}, ""ZeroCERT"": {""detected"": false, ""result"": ""clean site""}, ""Baidu-International"": {""detected"": false, ""result"": ""clean site""}}}, {""permalink"": ""https://www.virustotal.com/url/71f1c1fff731ae942a55743741a3aa7e09b6c2f7d62e37023814dc63e72596de/analysis/1444982173/"", ""resource"": ""http://dlce.antivir.com/package/wks_avira/win32/en/pecl/avira_antivir_personal_en.exe"", ""url"": ""http://dlce.antivir.com/package/wks_avira/win32/en/pecl/avira_antivir_personal_en.exe"", ""response_code"": 1, ""scan_date"": ""2015-10-16 07:56:13"", ""scan_id"": ""71f1c1fff731ae942a55743741a3aa7e09b6c2f7d62e37023814dc63e72596de-1444982173"", ""verbose_msg"": ""Scan finished, scan information embedded in this object"", ""filescan_id"": null, ""positives"": 1, ""total"": 65, ""scans"": {""CLEAN MX"": {""detected"": false, ""result"": ""clean site""}, ""Rising"": {""detected"": false, ""result"": ""clean site""}, ""OpenPhish"": {""detected"": false, ""result"": ""clean site""}, ""VX Vault"": {""detected"": false, ""result"": ""clean site""}, ""ZDB Zeus"": {""detected"": false, ""result"": ""clean site""}, ""AutoShun"": {""detected"": false, ""result"": ""unrated site""}, ""ZCloudsec"": {""detected"": false, ""result"": ""clean site""}, ""PhishLabs"": {""detected"": false, ""result"": ""unrated site""}, ""Zerofox"": {""detected"": false, ""result"": ""clean site""}, ""K7AntiVirus"": {""detected"": false, ""result"": ""clean site""}, "" CloudStat"": {""detected"": false, ""result"": ""clean site""}, ""SecureBrain"": {""detected"": false, ""result"": ""clean site""}, ""Quttera"": {""detected"": false, ""result"": ""clean site""}, ""AegisLab WebGuard"": {""detected"": false, ""result"": ""clean site""}, ""MalwareDomainList"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://www.malwaredomainlist.com/mdl.php?search=dlce.antivir.com""}, ""ZeusTracker"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""https://zeustracker.abuse.ch/monitor.php?host=dlce.antivir.com""}, ""zvelo"": {""detected"": false, ""result"": ""clean site""}, ""Google Safebrowsing"": {""detected"": false, ""result"": ""clean site""}, ""FraudScore"": {""detected"": false, ""result"": ""clean site""}, ""Kaspersky"": {""detected"": false, ""result"": ""clean site""}, ""BitDefender"": {""detected"": false, ""result"": ""clean site""}, ""Wepawet"": {""detected"": false, ""result"": ""clean site""}, ""G-Data"": {""detected"": false, ""result"": ""clean site""}, ""C-SIRT"": {""detected"": false, ""result"": ""clean site""}, ""CyberCrime"": {""detected"": false, ""result"": ""clean site""}, ""Websense ThreatSeeker"": {""detected"": true, ""result"": ""malicious site""}, ""MalwarePatrol"": {""detected"": false, ""result"": ""clean site""}, ""Webutation"": {""detected"": false, ""result"": ""clean site""}, ""Trustwave"": {""detected"": false, ""result"": ""clean site""}, ""Web Security Guard"": {""detected"": false, ""result"": ""clean site""}, ""ADMINUSLabs"": {""detected"": false, ""result"": ""clean site""}, ""Malwarebytes hpHosts"": {""detected"": false, ""result"": ""clean site""}, ""Dr.Web"": {""detected"": false, ""result"": ""clean site""}, ""AlienVault"": {""detected"": false, ""result"": ""clean site""}, ""Emsisoft"": {""detected"": false, ""result"": ""clean site""}, ""Malc0de Database"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://malc0de.com/database/index.php?search=dlce.antivir.com""}, ""SpyEyeTracker"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""https://spyeyetracker.abuse.ch/monitor.php?host=dlce.antivir.com""}, ""malwares.com URL checker"": {""detected"": false, ""result"": ""clean site""}, ""Phishtank"": {""detected"": false, ""result"": ""clean site""}, ""Malwared"": {""detected"": false, ""result"": ""clean site""}, ""Avira"": {""detected"": false, ""result"": ""clean site""}, ""StopBadware"": {""detected"": false, ""result"": ""unrated site""}, ""Antiy-AVL"": {""detected"": false, ""result"": ""clean site""}, ""SCUMWARE.org"": {""detected"": false, ""result"": ""clean site""}, ""FraudSense"": {""detected"": false, ""result"": ""clean site""}, ""Opera"": {""detected"": false, ""result"": ""clean site""}, ""Comodo Site Inspector"": {""detected"": false, ""result"": ""clean site""}, ""Malekal"": {""detected"": false, ""result"": ""clean site""}, ""ESET"": {""detected"": false, ""result"": ""clean site""}, ""Sophos"": {""detected"": false, ""result"": ""unrated site""}, ""Yandex Safebrowsing"": {""detected"": false, ""result"": ""clean site"", ""detail"": ""http://yandex.com/infected?l10n=en&url=http://dlce.antivir.com/package/wks_avira/win32/en/pecl/avira_antivir_personal_en.exe""}, ""Spam404"": {""detected"": false, ""result"": ""clean site""}, ""Malware Domain Blocklist"": {""detected"": false, ""result"": ""clean site""}, ""Blueliv"": {""detected"": false, ""result"": ""clean site""}, ""Netcraft"": {""detected"": false, ""result"": ""unrated site""}, ""PalevoTracker"": {""detected"": false, ""result"": ""clean site""}, ""CRDF"": {""detected"": false, ""result"": ""clean site""}, ""ThreatHive"": {""detected"": false, ""result"": ""clean site""}, ""ParetoLogic"": {""detected"": false, ""result"": ""clean site""}, ""Tencent"": {""detected"": false, ""result"": ""clean site""}, ""URLQuery"": {""detected"": false, ""result"": ""unrated site""}, ""Sucuri SiteCheck"": {""detected"": false, ""result"": ""clean site""}, ""Fortinet"": {""detected"": false, ""result"": ""clean site""}, ""ZeroCERT"": {""detected"": false, ""result"": ""clean site""}, ""Baidu-International"": {""detected"": false, ""result"": ""clean site""}}}]";
var deserializer = new RestSharp.Deserializers.JsonDeserializer();
var reports = deserializer.Deserialize<List<MyUrlReport>>(new RestSharp.RestResponse { Content = json });
```

**Comment by Genbox** [comment]
Thanks. I will take a look at it.

**Comment by Genbox** [comment]
Fixed in 6b3ccfbd248185cd4bda931c6476d3c4e43d35c3
